bundles/org.openhab.binding.lgthinq/src/main/java/org/openhab/binding/lgthinq/internal/api/TokenManager.java

```diff
@@ -0,0 +1,168 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.lgthinq.internal.api;
+
+import static org.openhab.binding.lgthinq.internal.LGThinQBindingConstants.*;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.openhab.binding.lgthinq.internal.errors.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * The {@link TokenManager} Principal facade to manage all token handles
+ *
+ * @author Nemer Daud - Initial contribution
+ */
+@NonNullByDefault
+public class TokenManager {
+    private static final int EXPIRICY_TOLERANCE_SEC = 60;
+    private static final Logger logger = LoggerFactory.getLogger(TokenManager.class);
+    private final OauthLgEmpAuthenticator oAuthAuthenticator;
+    private final ObjectMapper objectMapper = new ObjectMapper();
+    private final Map<String, TokenResult> tokenCached = new ConcurrentHashMap<>();
+
+    public TokenManager(HttpClient httpClient) {
+        oAuthAuthenticator = new OauthLgEmpAuthenticator(httpClient);
+    }
+
+    public boolean isTokenExpired(TokenResult token) {
+        Calendar c = Calendar.getInstance();
+        c.setTime(token.getGeneratedTime());
+        c.add(Calendar.SECOND, token.getExpiresIn() - EXPIRICY_TOLERANCE_SEC);
+        Date expiricyDate = c.getTime();
+        return expiricyDate.before(new Date());
+    }
+
+    @NonNull
+    public TokenResult refreshToken(String bridgeName, TokenResult currentToken) throws RefreshTokenException {
+        try {
+            TokenResult token = oAuthAuthenticator.doRefreshToken(currentToken);
+            objectMapper.writeValue(new File(getConfigDataFileName(bridgeName)), token);
+            return token;
+        } catch (IOException e) {
+            throw new RefreshTokenException("Error refreshing LGThinq token", e);
+        }
+    }
+
+    private String getConfigDataFileName(String bridgeName) {
+        return String.format(THINQ_CONNECTION_DATA_FILE, bridgeName);
+    }
+
+    public boolean isOauthTokenRegistered(String bridgeName) {
+        File tokenFile = new File(getConfigDataFileName(bridgeName));
+        // TODO - check if the file content is valid.
```

Comment:
Are you able to implement this?

Reply:
I don't think it's necessary. Actually this method only needs to check if the token file exits. In the sequence, the file is opened and parsed, and if the file has some invalid content, the error is reported by the getValidRegisteredToken method. I think this TODO is very old. I will remove it.